### PR TITLE
feat: filter inputs - string autocomplete to mantine

### PR DESCRIPTION
**Pull request by IrakliJani**
Closes: https://github.com/lightdash/lightdash/issues/6730

### Description:

CleanShot 2023-10-31 at 01 26 23@2x

CleanShot 2023-10-31 at 01 26 38@2x

CleanShot 2023-10-31 at 01 27 18@2x

CleanShot 2023-10-31 at 01 27 31@2x

### Reviewer actions

- [ ] I have manually tested the changes in the preview environment
- [ ] I have reviewed the code
- [ ] I understand that "request changes" will block this PR from merging

### CODE REVIEW

**Comment by IrakliJani** [comment]
@magnew I think remaining tests that fail are related to the chart config sidebar change. not sure why it's enabled here 🤔

**Comment by magnew** [comment]
> @magnew I think remaining tests that fail are related to the chart config sidebar change. not sure why it's enabled here 🤔

My bad. I turned it on to test something. Will disable and re-run.

File: packages/frontend/src/components/common/Filters/FilterInputs/AutoComplete/FilterMultiAutoComplete.tsx [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+        search,
+        initialSuggestionData,
+        projectUuid,
+        field,
+        autocompleteFilterGroup,
+        true,
+        { refetchOnMount: 'always' },
+    );
+
+    const results = useMemo(() => [...resultsSet], [resultsSet]);
```
Comment (magnew) [inline packages/frontend/src/components/common/Filters/FilterInputs/AutoComplete/FilterMultiAutoComplete.tsx:57]: No big deal, but does this useMemo do anything? It seems like it just keeps an updated copy of the results, but you could just use them directly, right? Maybe I am missing something and this is critical :D

File: packages/frontend/src/components/common/Filters/FilterInputs/AutoComplete/FilterMultiAutoComplete.tsx [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+    );
+
+    const handleOnClose = useCallback(() => {
+        setSearch('');
+    }, [setSearch]);
+
+    const data = useMemo(() => {
+        // Mantine does not show value tag if value is not found in data
+        // so we need to add it manually here
+        // also we are merging status indicator as a first item
```
Comment (magnew) [inline packages/frontend/src/components/common/Filters/FilterInputs/AutoComplete/FilterMultiAutoComplete.tsx:81]: Isn't this what onCreate is for? This probably works too, but i think we usually add them to the data in onCreate. Which is maybe a little clearer and doesnt need an extra memo.

**Review by magnew — APPROVED** [review]
Looks good! Left a couple non-blocking questions.

File: packages/frontend/src/components/common/Filters/FilterInputs/AutoComplete/FilterMultiAutoComplete.tsx (hunk 1 above)
Comment (IrakliJani) [reply packages/frontend/src/components/common/Filters/FilterInputs/AutoComplete/FilterMultiAutoComplete.tsx:57]: oh it returns a set not an array. I think it was like that before let me check

File: packages/frontend/src/components/common/Filters/FilterInputs/AutoComplete/FilterMultiAutoComplete.tsx (hunk 2 above)
Comment (IrakliJani) [reply packages/frontend/src/components/common/Filters/FilterInputs/AutoComplete/FilterMultiAutoComplete.tsx:81]: so, `data` prop is basically `options` - what options show up in the dropdown. and `value` is for an array of selected options from data. if we have 

data of = [1,2,3]
and values of = [2,3,4]
then it only displays 2,3 as tags and omits 4

hope this explanation makes sense

File: packages/frontend/src/components/common/Filters/FilterInputs/AutoComplete/FilterMultiAutoComplete.tsx (hunk 1 above)
Comment (magnew) [reply packages/frontend/src/components/common/Filters/FilterInputs/AutoComplete/FilterMultiAutoComplete.tsx:57]: No problem, it seems ok, just a little strange

File: packages/frontend/src/components/common/Filters/FilterInputs/AutoComplete/FilterMultiAutoComplete.tsx (hunk 2 above)
Comment (magnew) [reply packages/frontend/src/components/common/Filters/FilterInputs/AutoComplete/FilterMultiAutoComplete.tsx:81]: Ok, yea, I see. Thank you.

**Review by magnew — APPROVED** [review]

**Comment by IrakliJani** [comment]
@magnew I was doing one more round of testing and it turns out I forgot onPaste function which I just added. I also renamed component and did some cleanups

**Pull request merged**